cache_system/src/backend/policy/remove_if.rs

```diff
@@ -0,0 +1,196 @@
+//! Backend that supports custom removal / expiry of keys
+use metric::U64Counter;
+use parking_lot::Mutex;
+use std::{fmt::Debug, hash::Hash, marker::PhantomData, sync::Arc};
+
+use super::{CallbackHandle, ChangeRequest, Subscriber};
+
+/// Allows explicitly removing entries from the cache.
+#[derive(Debug, Clone)]
+pub struct RemoveIfPolicy<K, V>
+where
+    K: Clone + Eq + Debug + Hash + Ord + Send + 'static,
+    V: Clone + Debug + Send + 'static,
+{
+    // the policy itself doesn't do anything, the handles will do all the work
+    _phantom: PhantomData<(K, V)>,
+}
+
+impl<K, V> RemoveIfPolicy<K, V>
+where
+    K: Clone + Eq + Debug + Hash + Ord + Send + 'static,
+    V: Clone + Debug + Send + 'static,
+{
+    /// Create new policy.
+    pub fn create_constructor_and_handle(
+        name: &'static str,
+        metric_registry: &metric::Registry,
+    ) -> (
+        impl FnOnce(CallbackHandle<K, V>) -> Self,
+        RemoveIfHandle<K, V>,
+    ) {
+        let metric_removed_by_predicate = metric_registry
+            .register_metric::<U64Counter>(
+                "cache_removed_by_custom_condition",
+                "Number of entries removed from a cache via a custom condition",
+            )
+            .recorder(&[("name", name)]);
+
+        let handle1 = RemoveIfHandle {
+            callback_handle: Arc::new(Mutex::new(None)),
+            metric_removed_by_predicate,
+        };
+        let handle2 = handle1.clone();
```

Comment:
```suggestion
        let handle_captured = handle1.clone();
```

I found myself trying to figure out why there were two handles for a moment -- perhaps a slightly different name would make it clearer the second handle is for the closure

Reply:
done (also renamed `handle1`).